lib/sidekiq/cloudwatchmetrics.rb

```diff
@@ -45,7 +45,7 @@ class Publisher
 
     DEFAULT_INTERVAL = 60 # seconds
 
-    def initialize(config: Sidekiq, client: Aws::CloudWatch::Client.new, namespace: "Sidekiq", process_metrics: true, additional_dimensions: {}, interval: DEFAULT_INTERVAL)
+    def initialize(config: Sidekiq::Config.new, client: Aws::CloudWatch::Client.new, namespace: "Sidekiq", process_metrics: true, additional_dimensions: {}, interval: DEFAULT_INTERVAL)
```

Comment:
`Sidekiq::Config` was [introduced in sidekiq 7.0](https://github.com/sidekiq/sidekiq/pull/5487) so this is a backwards incompatible change. We can either make the change backwards compatible, or pull forward the minimum sidekiq version to 7.0. I'm not sure which I prefer.

What's Sidekiq's version support policy? We could probably match it.

Reply:
Sidekiq::Config.new isn't appropriate anyway - it should share the config with sidekiq to appropriately configure logging etc.

I'll pop on a couple of commits.

Reply:
I can't push to your branch. I will push to another.